### PR TITLE
#163303069 user login api endpoint

### PR DESCRIPTION
**Pull request by JosephNjuguna**
### What does this PR do??
this pr enables a user to sign in to their account
### Description of task to be completed??
- add sign in api endpoint
- inherit from other validations to check if  an email already exist or not 
- password validation to check if email is same as the on used to sign up

### How should this be tested?
clone this repo to your machine.
open the repo folder in terminal
add virtual env  `python3 -m virtualenv env`
export the APP environment ` export FLASK_ENV=development`
start the app by running following command `flask run`
open postman and navigate to following link `http://127.0.0.1:5000/api/v2/auth/login`
test the endpoint with this data:
```
{ 
	"email": "test1@mail.com",
    "password":"testpassword1"
	
}
```
### Relevat Pivotal tracker stories?
[pivotal tracker link ](https://www.pivotaltracker.com/story/show/163303069)
### SCREENHOTS AVAILABLE >
 
screenshot from 2019-01-23 11-03-15

### CODE REVIEW

File: app/api/v2/models/users_models.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ class UserModel(BaseModel):
     def __init__(self, firstname="firstname", lastname="lastname", email="@mail.com", password="pass",confirm_password="pass", phonenumber="1233", username="user"):
         """initialize the user model"""
         self.db = init_db()
-        #user data
+        """user data"""
```
Comment (oagutu) [inline app/api/v2/models/users_models.py:17]: This is not necessary(docstrings within the methods is enough) nor does it follow conventions with regard to comments. Please omit this as well as all similar instances within this module.

File: app/api/v2/models/users_models.py [hunk 2]
```diff
@@ -73,3 +73,18 @@ def add_user(self):
         database.commit()
         cur.close()
         return int(user_id)
+
+      #log in user
+    
+    """user sign in"""
+    def login_user(self, email):
+        """searches for a single user by use of email"""
+        """return user from the db given a username"""
```
Comment (oagutu) [inline app/api/v2/models/users_models.py:82]: All these can be under the same triple quotes. Refer to the [Google docstring conventions](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)s for help with this.

File: app/api/v2/models/users_models.py [hunk 3]
```diff
@@ -73,3 +73,18 @@ def add_user(self):
...
+        database = self.db
+        curr = database.cursor()
+        curr.execute(
+            """SELECT id, firstname, lastname, passwor,registered
```
Comment (oagutu) [inline app/api/v2/models/users_models.py:86]: minor typo: `password`

**Review by oagutu — COMMENTED** [review]
Good job. I noticed a few issues that need addressing:
- Non-descriptive error when signing up, ie. `”Expecting a field key”`. Therefore I can’t tell exactly what’s missing from my request data
- Error messages when either password invalid or email non-existent are not in JSON format

**Comment by JosephNjuguna** [comment]
@oagutu  thanks for the feedback I will work it on a bug branch. then  update you  when i raise pr for the bug branch

**Pull request merged**